File: app/schemas/monitor.py
"""监控Schema"""

from datetime import date, datetime
from typing import Any

from pydantic import BaseModel


class FactorHealthResponse(BaseModel):
    trade_date: date
    factor_name: str
    coverage_rate: float | None = None
    missing_rate: float | None = None
    ic_rolling: float | None = None
    ic_mean: float | None = None
    icir: float | None = None
    psi: float | None = None
    health_status: str = "healthy"

    class Config:
        from_attributes = True


class ModelHealthResponse(BaseModel):
    trade_date: date
    model_id: str
    prediction_drift: float | None = None
    feature_importance_drift: float | None = None
    oos_score: float | None = None
    health_status: str = "healthy"

    class Config:
        from_attributes = True


class PortfolioMonitorResponse(BaseModel):
    trade_date: date | None = None
    model_id: int | None = None
    industry_exposure: dict[str, float] | None = None
    style_exposure: dict[str, float] | None = None
    turnover_rate: float | None = None
    crowding_score: float | None = None


class LiveTrackingResponse(BaseModel):
    model_id: int | None = None
    execution_deviation: float | None = None
    cost_deviation: float | None = None
    drawdown: float | None = None
    fill_rate: float | None = None


class MonitorAlertResponse(BaseModel):
    alert_id: int
    alert_time: datetime | None = None
    alert_type: str | None = None
    severity: str | None = None
    object_type: str | None = None
    object_name: str | None = None
    message: str | None = None
    resolved_flag: bool = False

    class Config:
        from_attributes = True


class RegimeResponse(BaseModel):
    trade_date: str | None = None
    regime: str = "trending"
    confidence: float | None = None
    regime_detail: dict[str, Any] | None = None
    module_weight_adjustment: dict[str, float] | None = None

File: frontend/src/api/types/monitor.ts
export interface MonitorFactorHealth {
  trade_date: string;
  factor_name: string;
  factor_code?: string;
  category?: string;
  coverage_rate: number | null;
  missing_rate: number | null;
  ic_rolling: number | null;
  ic_mean: number | null;
  ir: number | null;
  psi: number | null;
  health_status: string;
}

export interface MonitorModelHealth {
  trade_date: string;
  model_id: string;
  prediction_drift: number | null;
  feature_importance_drift: number | null;
  oos_score: number | null;
  health_status: string;
}

export interface MonitorAlert {
  alert_id: number;
  alert_time: string | null;
  alert_type: string | null;
  severity: string | null;
  object_type: string | null;
  object_name: string | null;
  message: string | null;
  resolved_flag: boolean;
}

export interface Regime {
  trade_date: string | null;
  regime: string;
  confidence: number | null;
  regime_detail: Record<string, unknown> | null;
  module_weight_adjustment: Record<string, number> | null;
}

export interface PortfolioMonitor {
  trade_date: string | null;
  model_id: number | null;
  industry_exposure: Record<string, number> | null;
  style_exposure: Record<string, number> | null;
  turnover_rate: number | null;
  crowding_score: number | null;
}

export interface LiveTracking {
  model_id: number | null;
  execution_deviation: number | null;
  cost_deviation: number | null;
  drawdown: number | null;
  fill_rate: number | null;
}
